### PR TITLE
Sign the `MSIX` files for the store

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Sign the `MSIX` files for the store

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml [hunk 1]
```diff
@@ -38,12 +38,16 @@ jobs:
...
+
+      $msixFiles = @(
         "$(System.ArtifactsDirectory)\signed\PowerShell-$(Version)-win-x86.msix"
         "$(System.ArtifactsDirectory)\signed\PowerShell-$(Version)-win-x64.msix"
         "$(System.ArtifactsDirectory)\signed\PowerShell-$(Version)-win-arm32.msix"
         "$(System.ArtifactsDirectory)\signed\PowerShell-$(Version)-win-arm64.msix"
       )
-      tools/releaseBuild/generatePackgeSigning.ps1 -AuthenticodeFiles $authenticodeFiles -path "$(System.ArtifactsDirectory)\package.xml"
+
+      tools/releaseBuild/generatePackgeSigning.ps1 -AuthenticodeFiles $authenticodeFiles -path "$(System.ArtifactsDirectory)\package.xml" -MsixCertType $env:MSIX_TYPE -MsixFiles
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/windows-package-signing.yml:50]:
```suggestion
      tools/releaseBuild/generatePackgeSigning.ps1 -AuthenticodeFiles $authenticodeFiles -path "$(System.ArtifactsDirectory)\package.xml" -MsixCertType $env:MSIX_TYPE -MsixFiles $msixFiles
```

File: tools/releaseBuild/generatePackgeSigning.ps1 [hunk 2]
```diff
@@ -95,6 +98,23 @@ foreach ($file in $ThirdPartyFiles) {
...
 
+foreach ($file in $MsixFiles) {
+    switch($MsixCertType)
+    {
+        'release' {
+            $signType = 'CP-459155'
+        }
+        'preview' {
+            $signType = 'CP-459155'
+        }
```
Comment (adityapatwardhan) [inline tools/releaseBuild/generatePackgeSigning.ps1:109]: Both are same right?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/generatePackgeSigning.ps1 (hunk 2 above)
Comment (TravisEz13) [reply tools/releaseBuild/generatePackgeSigning.ps1]: resolved

**Review by SteveL-MSFT — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by ShadowXVII** [comment]
@TravisEz13 , still getting issues with PowerShell in Store-published apps. 
The system cannot find the file specified Microsoft.Management.Infrastructure.dll.

I can see the runtime in the app release folder, but the UWP packaging doesn't seem to pick it up and bundle it into the store package before submission: (i.e. win10-x64\lib\netstandard1.6\Microsoft.Management.Infrastructure.dll)

image

Not to sure if this is the best repo for this issue or whether it's more related to https://github.com/PowerShell/MMI/issues/45.

**Comment by TravisEz13** [comment]
@ShadowXVII Are you saying this is because it's unsigned?  Commenting on a closed issue isn't really a great idea unless you are sure it is releated.

**Comment by ShadowXVII** [comment]
@TravisEz13, well no, but if it's working for others then I must be missing something in my configuration. I've created an issue in MMI to see if they can assist.

**Pull request merged**